python.d/go_expvar.chart.py

```diff
@@ -0,0 +1,229 @@
+                    continue
+
+                if ev_key in self.expvars:
+                    self.info("Duplicate expvar key {}: skipping line.".format(ev_key))
+                    continue
+
+                self.expvars[ev_key] = ev_type
+
+                chart_dict['lines'].append(
+                    [
+                        line.get('id', ''),
+                        line.get('name', ''),
+                        line.get('algorithm', ''),
+                        line.get('multiplier', 1),
+                        line.get('divisor', 100 if ev_type == 'float' else 1),
+                        line.get('hidden', False)
+                    ]
+                )
+
+            self.order.append(chart_id)
+            self.definitions[chart_id] = chart_dict
+
+    def _get_data(self):
+        """
+        Format data received from http request
+        :return: dict
+        """
+
+        data = json.loads(self._get_raw_data())
+
+        expvars = dict()
+        if self.configuration.get('collect_memstats'):
+            expvars.update(self._parse_memstats(data))
+
+        if self.configuration.get('extra_charts'):
+            del (data['memstats'])
+            flattened = flatten(data)
+            for k, v in flattened.items():
+                key_type = self.expvars.get(k)
+                try:
+                    if key_type == 'int':
+                        expvars[k] = int(v)
+                    elif key_type == 'float':
+                        # if the value type is float, multiply it by 1000 and set line divisor to 1000
+                        expvars[k] = float(v) * 100
+                    else:
+                        # the key_type is None (not defined in config)
+                        continue
+                except ValueError:
+                    self.info("Failed to parse value for key {} as {}, ignoring key.".format(k, key_type))
+                    del self.expvars[k]
+
+        return expvars
+
+    @staticmethod
+    def _parse_memstats(data):
+
+        memstats = data['memstats']
+
+        # calculate the number of live objects in memory
+        live_objs = int(memstats['Mallocs']) - int(memstats['Frees'])
+
+        # calculate GC pause times average
+        # the Go runtime keeps the last 256 GC pause durations in a circular buffer,
+        #  so we need to filter out the 0 values before the buffer is filled
+        gc_pauses = memstats['PauseNs']
+        try:
+            gc_pause_avg = sum(gc_pauses) / len([x for x in gc_pauses if x > 0])
+        # no GC cycles have occured yet
+        except ZeroDivisionError:
+            gc_pause_avg = 0
+
+        heap_inuse = memstats['HeapInuse'] / 1024
+        heap_alloc = memstats['HeapAlloc'] / 1024
+        stack_inuse = memstats['StackInuse'] / 1024
+        mspan_inuse = memstats['MSpanInuse'] / 1024
+        mcache_inuse = memstats['MCacheInuse'] / 1024
```

Comment:
Yes, I agree. Fixed in [b94c9c9ae](https://github.com/firehol/netdata/pull/2280/commits/b94c9c9aeea64e6b87a34d6e8b5c48bf88970ac3)

I still use division at [line 211](https://github.com/firehol/netdata/pull/2280/files#diff-216da685dc9d4636c2eb947a55aa9ea6R211), so I added a future import. In this case I could probably get by with integer division, as the values will be in the tens of thousands minimum, so the decimal part is not really needed and netdata ignores it anyway.